src/Models/Sources.php

```diff
@@ -102,4 +103,42 @@ public function validateAppleUser(string $identityToken): object
 
         return $appleUserInfo;
     }
+
+    /**
+     * validation.
+     *
+     * @param  string $email
+     * @param  string $token
+     *
+     * @return bool
+     */
+    public function validation(string $email, string $token) : bool
+    {
+        $di = DI::getDefault();
+        try {
+            switch ($this->title) {
+                case 'google':
+                        $client = $di->getGoogle();
+                        $payload = $client->verifyIdToken($token);
+                        if ($payload) {
+                            $userid = $payload['sub'];
+                            return $payload['email'] === $email;
```

Comment:
@FredPeal why do you validate the email here but not on fb?

Reply:
Because I can't test with Facebook login on development mode. For test, the Facebook app must be prod